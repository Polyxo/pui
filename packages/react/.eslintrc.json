{
  "root": true,
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint"],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:storybook/recommended"
  ],
  "rules": {
    "no-unused-vars": "off",
    "no-console": "error",
    "@typescript-eslint/no-unused-vars": ["error"]
  }
}
