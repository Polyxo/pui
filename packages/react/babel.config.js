module.exports = {
  presets: [
    "@babel/preset-env",
    "@babel/preset-react",
    "@babel/preset-typescript",
  ],
  plugins: [
    "dev-expression",
    "@babel/plugin-syntax-dynamic-import",
    "@babel/plugin-syntax-import-meta",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-export-namespace-from",
    "@babel/plugin-proposal-export-default-from",
  ],
};
