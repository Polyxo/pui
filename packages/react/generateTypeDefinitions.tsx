import { rm, stat, readdir, writeFile, mkdir } from 'fs/promises';
import path from 'path';
import { parse } from 'react-docgen-typescript';

const distUrl = path.resolve(__dirname, '../../website/types/');

const options = {
  savePropValueAsString: true,
  propFilter: (prop, component) => {
    if (prop.declarations !== undefined && prop.declarations.length > 0) {
      const hasPropAdditionalDescription = prop.declarations.find(
        (declaration) => {
          return !declaration.fileName.includes('node_modules');
        }
      );

      return Boolean(hasPropAdditionalDescription);
    }

    return true;
  },
};

export default async function walk(directory: string) {
  let fileList: string[] = [];

  const files = await readdir(directory);
  for (const file of files) {
    const p = path.join(directory, file);
    if ((await stat(p)).isDirectory()) {
      fileList = [...fileList, ...(await walk(p))];
    } else {
      fileList.push(p);
    }
  }

  const tsxFileList = fileList.filter(
    (file) => file.endsWith('.tsx') && !file.includes('stories')
  );

  await Promise.all(
    tsxFileList.map(async (file) => {
      const propTypes = JSON.parse(JSON.stringify(parse(file, options)));

      try {
        await mkdir(path.dirname(path.join(distUrl, file)), {
          recursive: true,
        });
      } catch (error) {
        console.log('collectTypes error', error);
      }

      console.log(
        'propTypes',
        path.join(distUrl, file.replace('.tsx', '.json'))
      );
      await writeFile(
        path.join(distUrl, file.replace('.tsx', '.json')),
        JSON.stringify(propTypes, null, 2)
      );
      return null;
    })
  );

  return fileList;
}

async function startGenerateTypes() {
  await rm(distUrl, { recursive: true, force: true });

  await walk('./src');
  process.exit();
}

startGenerateTypes();
