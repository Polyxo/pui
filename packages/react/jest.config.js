module.exports = {
  preset: "ts-jest",
  testEnvironment: "jsdom",
  transform: {
    "^.+\\.tsx?$": "ts-jest",
  },

  roots: ["<rootDir>/src"],
  testMatch: [
    //"**/__tests__/**/*.+(ts|tsx|js)",
    "**/?(*.)+(spec|test).+(ts|tsx|js)",
  ],
};
