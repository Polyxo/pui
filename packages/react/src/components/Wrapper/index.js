export Wrapper from './Wrapper';
// export { pageWidths } from './Wrapper';
