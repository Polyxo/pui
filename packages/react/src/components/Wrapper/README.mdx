Wrapper can wrap it's content in a specific width depending on if the application has a full or fixed width.

```js
import { Wrapper } from "@progressiveui/react";
```

```js
<Wrapper pageWidth="xs" mobilePageWidth="full">
  Content
</Wrapper>
```
