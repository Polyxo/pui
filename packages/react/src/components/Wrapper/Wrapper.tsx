import * as React from "react";
import classNames from "classnames";
import useSettings from "../../hooks/useSettings";
import { ScreenSize } from "../../utils";

export const pageWidths = ["sm", "md", "lg", "full"];

/** Wrapper can wrap it's content in a specific width depending on if the application has a full or fixed width. */
export interface WrapperProps extends React.HTMLAttributes<HTMLSpanElement> {
  /** Defines the width of the wrapper. Can be one of 'xs', 'sm', 'md', 'lg', 'full', 'narrow', 'narrower', or 'narrowest'. */
  pageWidth?:
    | "xs"
    | "sm"
    | "md"
    | "lg"
    | "full"
    | "narrow"
    | "narrower"
    | "narrowest";

  /** Specifies the page width for mobile devices. It uses the ScreenSize type. */
  mobilePageWidth?: ScreenSize;

  /** Determines the spacing around the wrapper. Can be 'md' for medium or 'xl' for extra large. */
  spacing?: "md" | "xl";

  /** Sets the background theme. Can be 'lighter' for a light background or 'dark' for a dark background. */
  background?: "lighter" | "dark";

  /** Additional CSS class names to be applied to the wrapper. */
  className?: string;

  /** CSS class names specifically for the background of the wrapper. */
  backgroundClassName?: string;

  /** Custom CSS properties for the background styling. */
  backgroundStyle?: React.CSSProperties;
}

const Wrapper: React.FC<WrapperProps> = (props) => {
  const { prefix } = useSettings();

  const {
    background,
    backgroundClassName,
    backgroundStyle,
    children,
    className,
    pageWidth,
    mobilePageWidth,
    spacing,
    ...other
  } = props;
  const wrapperClasses = classNames({
    [`${prefix}--wrapper`]: true,
    [`${prefix}--wrapper--width-lg`]: pageWidth === "narrow",
    [`${prefix}--wrapper--width-md`]: pageWidth === "narrower",
    [`${prefix}--wrapper--width-sm`]: pageWidth === "narrowest",
    [`${prefix}--wrapper--width-xs`]: pageWidth === "narrowest",
    [`${prefix}--wrapper--width-${pageWidth}`]: pageWidth,
    [`${prefix}--wrapper--width-mobile-full`]: mobilePageWidth === "full",
    [`${prefix}--wrapper--spacing-md`]: spacing === "md",
    [`${prefix}--wrapper--spacing-xl`]: spacing === "xl",
    [`${className}`]: className,
  });
  if (background || backgroundStyle || backgroundClassName) {
    const backgroundClasses = classNames(backgroundClassName, {
      [`${prefix}--wrapper--background-${background}`]: background,
    });

    return (
      <div className={backgroundClasses} style={backgroundStyle}>
        <div className={wrapperClasses} {...other}>
          {children}
        </div>
      </div>
    );
  }
  return (
    <div className={wrapperClasses} {...other}>
      {children}
    </div>
  );
};

export default Wrapper;
