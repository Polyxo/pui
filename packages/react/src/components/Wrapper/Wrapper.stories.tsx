import React from "react";
import Wrapper from ".";

export default {
  title: "Components/Content Related/Wrapper",
  component: Wrapper,
  parameters: {
    componentSubtitle: "Component",
    status: "released",
  },
};

export const WrapperDefault = (args) => (
  <Wrapper {...args}>
    Assisting 91.4 million people in around 83 countries each year, the World
    Food Programme (WFP) is the leading humanitarian organization saving lives
    and changing lives, delivering food assistance in emergencies and working
    with communities to improve nutrition and build resilience. As the
    international community has committed to end hunger, achieve food security
    and improved nutrition by 2030, one in nine people worldwide still do not
    have enough to eat. Food and food-related assistance lie at the heart of the
    struggle to break the cycle of hunger and poverty. On any given day, WFP has
    5,000 trucks, 20 ships and 92 planes on the move, delivering food and other
    assistance to those in most need.
  </Wrapper>
);
