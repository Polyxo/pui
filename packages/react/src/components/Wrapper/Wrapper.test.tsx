// import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Wrapper from "./Wrapper";

describe("Wrapper Component", () => {
  // Test basic rendering
  it("renders the Wrapper with content", () => {
    render(<Wrapper>Content</Wrapper>);
    expect(screen.getByText("Content")).toBeInTheDocument();
  });

  // Test different page widths
  it.each(["xs", "sm", "md", "lg", "full", "narrow", "narrower", "narrowest"])(
    "renders with specified page width '%s'",
    (pageWidth: any) => {
      render(<Wrapper pageWidth={pageWidth}>Content</Wrapper>);

      expect(screen.getByText("Content")).toHaveClass(
        `wfp--wrapper--width-${pageWidth}`
      );
    }
  );

  // Test mobile page width
  it.each(["sm", "md", "lg", "full"])(
    "renders with specified mobile page width '%s'",
    (mobilePageWidth: any) => {
      render(<Wrapper mobilePageWidth={mobilePageWidth}>Content</Wrapper>);
      if (mobilePageWidth === "full") {
        expect(screen.getByText("Content")).toHaveClass(
          `wfp--wrapper--width-mobile-full`
        );
      }
    }
  );

  // Test spacing
  it.each(["md", "xl"])(
    "renders with specified spacing '%s'",
    (spacing: any) => {
      render(<Wrapper spacing={spacing}>Content</Wrapper>);
      expect(screen.getByText("Content")).toHaveClass(
        `wfp--wrapper--spacing-${spacing}`
      );
    }
  );

  // Test background themes
  it.each(["lighter", "dark"])(
    "renders with specified background theme '%s'",
    (background: any) => {
      render(<Wrapper background={background}>Content</Wrapper>);
      expect(screen.getByText("Content")?.parentNode).toHaveClass(
        `wfp--wrapper--background-${background}`
      );
    }
  );

  // Test custom class name
  it("applies custom class names", () => {
    const customClass = "my-custom-class";
    render(<Wrapper className={customClass}>Content</Wrapper>);
    expect(screen.getByText("Content")).toHaveClass(customClass);
  });

  // Test custom background class name
  it("applies custom background class names", () => {
    const backgroundClass = "my-background-class";
    render(
      <Wrapper background="dark" backgroundClassName={backgroundClass}>
        Content
      </Wrapper>
    );
    expect(screen.getByText("Content")?.parentNode).toHaveClass(
      backgroundClass
    );
  });

  // Test custom background styles
  it("applies custom background styles", () => {
    const backgroundStyle = { backgroundColor: "red" };
    render(<Wrapper backgroundStyle={backgroundStyle}>Content</Wrapper>);
    expect(screen.getByText("Content")?.parentNode).toHaveStyle(
      backgroundStyle
    );
  });

  // Snapshot test
  it("matches snapshot", () => {
    const { asFragment } = render(<Wrapper>Snapshot Wrapper</Wrapper>);
    expect(asFragment()).toMatchSnapshot();
  });
});
