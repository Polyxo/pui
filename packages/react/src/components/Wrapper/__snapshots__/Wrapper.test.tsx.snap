// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="wfp--wrapper"
  >
    Snapshot Wrapper
  </div>
</DocumentFragment>
`;
